### PR TITLE
Allow setting default AutoMinorLocator 

### LINKED ISSUE #14233: Feature Request: Allow setting default AutoMinorLocator
Via `rcParams` you can turn on minor ticks/grid for all your plots but there is currently no way to give a default interval for minor ticks to be shown.

AutoMinorLocator [defaults to 4 or 5](https://matplotlib.org/_modules/matplotlib/ticker.html#AutoMinorLocator) so potentially it could check to see if this value is set in rcParams and use that first instead?

### PR DESCRIPTION
**Pull request by IlievskiV**
## PR Summary

Pull request to resolve #14233 - Allow setting default AutoMinorLocator.
This PR is related to PR #16762 which is stale for a while and seems to abandoned by the developer. 
The behavior is similar to what is described in PR #16762 and Issue #14233 .
In this PR I include: 
- Validator of for the non-negative integers
- Validator for the new `rcParams` option `ndivs` for the number of minor ticks
- Automated tests for the new options

## PR Checklist

- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [x] New features are documented, with examples if plot related.
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [x] Conforms to Matplotlib style conventions (install `flake8-docstrings` and `pydocstyle<4` and run `flake8 --docstring-convention=all`).
- [N/A] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [N/A] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Comment by IlievskiV** [comment]
Is there anyone to review?

File: lib/matplotlib/rcsetup.py
Comment (jklymak) [inline lib/matplotlib/rcsetup.py:797]: This isn't quite correct, is it?  Its only if its a string, right?

File: lib/matplotlib/ticker.py
Comment (jklymak) [inline lib/matplotlib/ticker.py:2888]:
```suggestion
        If *n* is set to 'auto', it will be set to 4 or 5.
```

Comment (jklymak) [reply lib/matplotlib/ticker.py:2888]: The "4 or 5" is not your fault, but it would be nice if we could avoid being so vague.  What does 4 or 5 depend on?

**Comment by jklymak** [comment]
overall this looks great, just a couple of nitpicks.  Please continue ping us if no one follows up...

**Review by jklymak — APPROVED** [review]

**Comment by jklymak** [comment]
Very nice.  We need a second review, so again don't hesitate to ping if no one gets to it in a couple of days.

File: lib/matplotlib/ticker.py
Comment (QuLogic) [inline lib/matplotlib/ticker.py:2916]: Does this apply to 3D plots? If so, it usually treats z the same as x, while this does the opposite.

Comment (IlievskiV) [reply lib/matplotlib/ticker.py:2916]: ah didn’t know. So maybe it could be `if self.axis.axis_name == 'x' or self.axis.axis_name == 'z'` ?

File: lib/matplotlib/tests/test_ticker.py
Comment (IlievskiV) [reply lib/matplotlib/tests/test_ticker.py:199]: I think it is from my editor, but I can manually remove the two indents and put only 1.

File: lib/matplotlib/ticker.py
Comment (QuLogic) [reply lib/matplotlib/ticker.py:2916]: Maybe just check 'y' first? Though leave a comment about it.

File: lib/matplotlib/tests/test_ticker.py
Comment (QuLogic) [reply lib/matplotlib/tests/test_ticker.py:199]: I would definitely squash these together:
```python
    @pytest.mark.parametrize('lim, ref', [
        ((0, 1.39),
         [0.05, 0.1, 0.15, 0.25, 0.3, 0.35, 0.45, 0.5, 0.55, 0.65, 0.7, 0.75,
          0.85, 0.9, 0.95, 1.05, 1.1, 1.15, 1.25, 1.3, 1.35]),
        ((0, 0.139),
         [0.005, 0.01, 0.015, 0.025, 0.03, 0.035, 0.045, 0.05, 0.055, 0.065,
          0.07, 0.075, 0.085, 0.09, 0.095, 0.105, 0.11, 0.115, 0.125, 0.13,
          0.135]),
    ])
```
and the second group could fit all parameters on a single line, in a similar fashion.

I'd also suggest combining the rcparam and argument version using a parametrization, something like `@pytest.mark.parametrize('use_rcparam', [False, True])` and:
```python
        if use_rcparam:
            context = {'xtick.minor.ndivs': n, 'ytick.minor.ndivs': n}
            kwargs = {}
        else:
            context = {}
            kwargs = {'n': n}
```
then passing those in the respective functions.

File: lib/matplotlib/tests/test_ticker.py
Comment (QuLogic) [inline lib/matplotlib/tests/test_ticker.py:196]: You should stack the `use_rcparam` parametrization so you don't have to write out `lim, ref` values twice.
```suggestion
    @pytest.mark.parametrize('use_rcparam', [False, True])
    @pytest.mark.parametrize(
        'lim, ref', [
```

**Comment by IlievskiV** [comment]
@QuLogic is it fine now, or I have to update or add something else?

**Comment by QuLogic** [comment]
This looks good, but one thing I should have asked earlier; can you add a [What's new? entry](https://matplotlib.org/devdocs/users/next_whats_new/README.html) for this?

**Comment by QuLogic** [comment]
Ping?

**Comment by IlievskiV** [comment]
Here I am, I forgot about this :) Will add the documentation soon.

**Comment by QuLogic** [comment]
Ping again? We'd like to make 3.4 soonish, this month or so.

**Comment by jklymak** [comment]
Lets move to 3.5.  If we hear from the author in the next day or so we can move back...

**Comment by QuLogic** [comment]
I've rebased and added a what's new note; might want to squash to fewer commits later.

**Comment by tacaswell** [comment]
Thank you @IlievskiV , congratulations on your first merged Matplotib PR 🎉 and apologies this took so long to make it through the review process.

I hope we hear from you again.

**Pull request merged**
